lib/executables/validate.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-use-before-define */
+const assertion = require('assert');
+const fs = require('fs');
+const path = require('path');
+const utils = require('../helpers/utils');
+const { print } = require('../helpers/log');
+
+const VIEW_CLASSES = [
+  'TextFieldView', 'TextAreaView', 'CheckBoxView', 'SelectView', 'MultiSelectView', 'PasswordFieldView', 'OAuthFieldView',
```

Comment:
There is also [`JSONataView`](https://github.com/elasticio/configuration-component/blob/master/component.json#L9), [`HTTPAuthView`](https://github.com/elasticio/rest-api-component/blob/master/component.json#L10) (credentials only), [WebhookAuthView](WebhookAuthView) (credentials only) and a couple of other component specific views.

Reply:
I actually didn't include these on purpose - my thought process was just that they are specialized views with unique purposes and for example if new ones are added in the future we won't have caught them all, but all it does is print a warning that the existing view is not part of the typical allowed viewsets, which a developer might want to be aware of.

I also wasn't sure if they are available as valid options for all components, or where to get a complete list? I have also seen RestAPIView as well, there are probably more?

Reply:
https://github.com/elasticio/elasticio/blob/2bf22c0add09e38d09aea6fe53c8004f3fa2a514/src/frontend/src/modules/react/fields/index.jsx